app/controllers/single_page_subscriptions_controller.rb

```diff
@@ -0,0 +1,58 @@
+class SinglePageSubscriptionsController < ApplicationController
+  include AccountHelper
+  include GovukPersonalisation::ControllerConcern
+
+  before_action do
+    head :not_found unless govuk_account_auth_enabled?
+  end
+
+  def show
+    content_item = GdsApi.content_store.content_item(topic).to_h
+    if logged_in?
+      subscriber = GdsApi.email_alert_api.link_subscriber_to_govuk_account(
+        govuk_account_session: @account_session_header,
+      )
+
+      subscriber_list = GdsApi.email_alert_api.find_or_create_subscriber_list(
+        {
+          url: topic,
+          name: content_item["title"],
+          content_id: content_item["content_id"],
+        },
+      )
+
+      GdsApi.email_alert_api.subscribe(
+        subscriber_list_id: subscriber_list.dig("subscriber_list", "id"),
+        address: subscriber.dig("subscriber", "address"),
+        frequency: "daily",
+        skip_confirmation_email: true,
+      )
+
+      redirect_to topic
```

Comment:
You'll want to stick a success flash in the session here, so we can show it on the page.  There's a `AccountSubscriptionsController::SUCCESS_FLASH` constant you can use.

In #1170 I introduce a `CreateAccountSubscriptionService` to remove some repeated code around this sort of thing, but I can refactor this file to use it as part of that PR.

Reply:
Ok, so now that #1170 is merged, you'll want to use `CreateAccountSubscriptionService` in this controller.